src/barraylabels.mli

```diff
@@ -1,366 +0,0 @@
-(** [ArrayLabels.to_list a] returns the list of all the elements of [a]. *)
-
-val of_list : 'a list -> 'a array
-(** [ArrayLabels.of_list l] returns a fresh array containing the elements
-    of [l]. *)
-
-val mapi : f:(int -> 'a -> 'b) -> 'a array -> 'b array
-(** Same as {!ArrayLabels.map}, but the
-    function is applied to the index of the element as first argument,
-    and the element itself as second argument. *)
-
-val fold_left : f:('a -> 'b -> 'a) -> init:'a -> 'b array -> 'a
-(** [ArrayLabels.fold_left f x a] computes
-    [f (... (f (f x a.(0)) a.(1)) ...) a.(n-1)],
-    where [n] is the length of the array [a]. *)
-
-val fold_right : f:('b -> 'a -> 'a) -> 'b array -> init:'a -> 'a
-(** [ArrayLabels.fold_right f a x] computes
-    [f a.(0) (f a.(1) ( ... (f a.(n-1) x) ...))],
-    where [n] is the length of the array [a]. *)
-
-
-(** {6 Sorting} *)
-
-
-val sort : cmp:('a -> 'a -> int) -> 'a array -> unit
-(** Sort an array in increasing order according to a comparison
-    function.  The comparison function must return 0 if its arguments
-    compare as equal, a positive integer if the first is greater,
-    and a negative integer if the first is smaller (see below for a
-    complete specification).  For example, {!Pervasives.compare} is
-    a suitable comparison function, provided there are no floating-point
-    NaN values in the data.  After calling [ArrayLabels.sort], the
-    array is sorted in place in increasing order.
-    [ArrayLabels.sort] is guaranteed to run in constant heap space
-    and (at most) logarithmic stack space.
-    The current implementation uses Heap Sort.  It runs in constant
-    stack space.
-    Specification of the comparison function:
-    Let [a] be the array and [cmp] the comparison function.  The following
-    must be true for all x, y, z in a :
-    -   [cmp x y] > 0 if and only if [cmp y x] < 0
-    -   if [cmp x y] >= 0 and [cmp y z] >= 0 then [cmp x z] >= 0
-    When [ArrayLabels.sort] returns, [a] contains the same elements as before,
-    reordered in such a way that for all i and j valid indices of [a] :
-    -   [cmp a.(i) a.(j)] >= 0 if and only if i >= j
-*)
-
-val stable_sort : cmp:('a -> 'a -> int) -> 'a array -> unit
-(** Same as {!ArrayLabels.sort}, but the sorting algorithm is stable (i.e.
-    elements that compare equal are kept in their original order) and
-    not guaranteed to run in constant heap space.
-    The current implementation uses Merge Sort. It uses [n/2]
-    words of heap space, where [n] is the length of the array.
-    It is usually faster than the current implementation of {!ArrayLabels.sort}.
-*)
-
-val fast_sort : cmp:('a -> 'a -> int) -> 'a array -> unit
-(** Same as {!ArrayLabels.sort} or {!ArrayLabels.stable_sort}, whichever is
-    faster on typical input.
-*)
-
-
-(**/**)
-
-(** {6 Undocumented functions} *)
-
-(* The following is for system use only. Do not call directly. *)
-
-external unsafe_get : 'a array -> int -> 'a = "%array_unsafe_get"
```

Comment:
I would discourage exposing this in the API personally. There are are uses for unsafe access, but at that point, I'd assume any user would be advance enough to provide their own implementation around the ffi interface.